core/runtime/runtime.cpp

```diff
@@ -31,13 +33,13 @@ c10::optional<RTDevice> get_most_compatible_device(const RTDevice& target_device
     if (device.device_name == target_device.device_name) {
       // First priority is selecting a candidate which agrees with the current device ID
       // If such a device is found, we can select it and break out of the loop
-      if (device.id == current_device.id && best_match.id != current_device.id) {
+      if (device.id == current_device.id) {
```

Comment:
Why are we changing these?

Reply:
During review, it was determined that these are not actually necessary conditions. Since the ID of each device is unique on a given machine, it should not be possible that `best_match.id == current_device.id`, so the check is not needed here.